src/server/auth/server/auth.go

```diff
@@ -381,24 +416,85 @@ func (a *apiServer) GetACL(ctx context.Context, req *authclient.GetACLRequest) (
 	if !a.isActivated() {
 		return nil, authclient.NotActivatedError{}
 	}
+
+	// Validate request
+	if req.Repo == nil || req.Repo.Name == "" {
+		return nil, fmt.Errorf("invalid request: must provide name of repo you want to modify")
+	}
+
+	// Get calling user
 	user, err := a.getAuthenticatedUser(ctx)
 	if err != nil {
 		return nil, err
 	}
 
 	// Read repo ACL from etcd
 	resp = &authclient.GetACLResponse{
-		Acl: &authclient.ACL{},
+		ACL: &authclient.ACL{},
 	}
-	err = a.acls.ReadOnly(ctx).Get(req.Repo.Name, resp.Acl)
+	err = a.acls.ReadOnly(ctx).Get(req.Repo.Name, resp.ACL)
 	if err != nil {
 		return nil, err
 	}
 	// For now, require READER access to read repo metadata (commits, and ACLs)
-	if resp.Acl.Entries == nil && !user.Admin {
-		return nil, fmt.Errorf("you must have at least reader access to %s to read its ACL", req.Repo.Name)
-	} else if resp.Acl.Entries != nil && resp.Acl.Entries[user.Username] < authclient.Scope_READER {
-		return nil, fmt.Errorf("you must have at least reader access to %s to read its ACL", req.Repo.Name)
+	if resp.ACL.Entries == nil && !user.Admin {
+		return nil, fmt.Errorf("you must have at least READER access to %s to read its ACL", req.Repo.Name)
+	} else if resp.ACL.Entries != nil && resp.ACL.Entries[user.Username] < authclient.Scope_READER {
+		return nil, fmt.Errorf("you must have at least READER access to %s to read its ACL", req.Repo.Name)
+	}
+	return resp, nil
+}
+
+func (a *apiServer) SetACL(ctx context.Context, req *authclient.SetACLRequest) (resp *authclient.SetACLResponse, retErr error) {
+	func() { a.Log(req, nil, nil, 0) }()
+	defer func(start time.Time) { a.Log(req, resp, retErr, time.Since(start)) }(time.Now())
+	if !a.isActivated() {
+		return nil, authclient.NotActivatedError{}
+	}
+
+	// Validate request
+	if req.Repo == nil || req.Repo.Name == "" {
+		return nil, fmt.Errorf("invalid request: must provide name of repo you want to modify")
+	}
+
+	// Get calling user
+	user, err := a.getAuthenticatedUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Read repo ACL from etcd
+	var authzErr error
+	_, rwErr := col.NewSTM(ctx, a.etcdClient, func(stm col.STM) error {
+		acls := a.acls.ReadWrite(stm)
+		if err != nil {
+			return err
+		}
+		var acl authclient.ACL
+		if err := acls.Get(req.Repo.Name, &acl); err != nil {
+			return err
+		}
+
+		// Require OWNER access to modify repo ACL
+		if acl.Entries == nil && !user.Admin {
+			authzErr = fmt.Errorf("you must have OWNER access to %s to modify its ACL", req.Repo.Name)
+			return nil
+		} else if acl.Entries != nil && acl.Entries[user.Username] < authclient.Scope_OWNER {
```

Comment:
Not that big of a deal, but since these two `if` clauses have the same code, we could get rid of the code duplication by just `or`-ing the two boolean expressions.